crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -1099,11 +1122,13 @@ impl RoomInfo {
         if !summary.is_empty() {
             if !summary.heroes.is_empty() {
                 // Parse the user IDs from Ruma.
-                self.summary.heroes_user_ids = summary
-                    .heroes
+                self.summary.room_heroes = summary
+                    .heroes // TODO: Load the profiles?
```

Comment:
From what I can tell (please correct me if I'm wrong), because we're in the `RoomInfo` here, it isn't possible to load profiles for these users before storing them?

Maybe it would make sense to make `RoomHero` an enum in this instance with `user_id` and `user_profile` being the 2 fields so we can tell whether or not to load the profiles when computing the room name?

Reply:
Alternatively, perhaps we could add another field `needs_profile` or similar to the struct and use that when computing the name, that might work better actually 🤔

Reply:
Why is not sufficient to have a `RoomHero` with only the user id set?

Reply:
Nothing other than it seems weird to me that there's no obvious explanation for the storage containing
- Heroes with their profile when using Sliding Sync
- Heroes without their profile when using non-sliding sync.

I.e. when making a call to  `room.heroes()` you can't trust the name/avatar values and have to decide whether to load the profiles based upon if you're using sliding sync or not.

Reply:
This only reflects a difference between sliding sync (will return avatar and display names for heroes by default) and non-sliding sync (will only return user ids). We could try to erase the difference here, but that would probably have to be happening on the receiving side, that is, when calling `Room::heroes()`, we oucld try to fill in the display name and the avatar URL only if it's missing.

Reply:
Yeah that does make sense, although I was very much hoping to keep `Room::heroes` sync.

Reply:
What do you plan to do with this TODO?

Reply:
I'm happy to simply remove it given it shouldn't affect us on EX (although that might not be the answer you were looking for). Longer term, filling in the missing profiles would work, so long as we could keep some form of sync access (I keep wondering if we could store `RoomInfo` on `Room` and have it automatically update it properties when anything changes).

Reply:
I've rebased and removed it.